build/extend/index.md

```diff
@@ -0,0 +1,64 @@
+- Detect and skip executing unused build stages
+- Parallelize building independent build stages
+- Incrementally transfer only the changed files in your build context between builds
+- Detect and skip transferring unused files in your build context
+- Use [Dockerfile frontend](dockerfile-frontend.md) implementations with many new features
+- Avoid side effects with rest of the API (intermediate images and containers)
+- Prioritize your build cache for automatic pruning
+
+Apart from many new features, the main areas BuildKit improves on the current
+experience are performance, storage management, and extensibility. From the
+performance side, a significant update is a new fully concurrent build graph
+solver. It can run build steps in parallel when possible and optimize out
+commands that don't have an impact on the final result. We have also optimized
+the access to the local source files. By tracking only the updates made to these
+files between repeated build invocations, there is no need to wait for local
+files to be read or uploaded before the work can begin.
+
+## LLB
+
+At the core of BuildKit is a new low-level build definition format called [LLB](https://github.com/moby/buildkit#exploring-llb){:target="_blank" rel="noopener" class="_"}
+(low-level builder). This is an intermediate binary format that end users are
+not exposed to but allows to easily build on top of BuildKit. LLB defines a
+content-addressable dependency graph that can be used to put together very
+complex build definitions. It also supports features not exposed in Dockerfiles,
+like direct data mounting and nested invocation.
+
+![Directed acyclic graph (DAG)](../images/buildkit-dag.svg){:class="invertible" style="width:60%"}
+
+Everything about execution and caching of your builds is defined only in LLB.
+The caching model is entirely rewritten compared to the current Dockerfile
+builder. Instead of doing heuristics to compare images, it can directly track
+the checksums of build graphs and content mounted to specific operations. This
+makes it much faster, more precise and portable. The build cache can even be
+exported to a registry, where it can be pulled on-demand by subsequent
+invocations on any host.
+
+LLB can be generated directly using a [golang client package](https://pkg.go.dev/github.com/moby/buildkit/client/llb)
+that allows defining the relationships between your build operations using Go
+language primitives. This gives you full power to run anything you can imagine,
+but will probably not be how most people will define their builds. Instead, we
+expect most users to use a frontend component or LLB nested invocation to run
+some prepared combination of build steps.
+
+A frontend is a component that takes a human-readable
+build format and converts it to LLB so BuildKit can execute it. Frontends can
+be distributed as images, and the user can target a specific version of a
+frontend that is guaranteed to work for the features used by their definition.
```

Comment:
for later: I think we will want to lift the description of "what's a BuildKit frontend" to a more prominent/visible place. I was confused about why it was called frontends until I read these paragraphs, and now it totally makes sense. Let's leave it here for now though!